packages/trpc/server/profile-router/router.ts

```diff
@@ -31,6 +35,36 @@ export const profileRouter = router({
     }
   }),
 
+  getUserByEmail: procedure
+    .input(ZRetrieveUserByEmailMutationSchema)
+    .mutation(async ({ input }) => {
+      try {
+        const { email } = input;
+
+        return await getUserByEmail({ email });
+      } catch (err) {
+        throw new TRPCError({
+          code: 'BAD_REQUEST',
+          message: 'We were unable to retrieve the specified account. Please try again.',
+        });
+      }
+    }),
```

Comment:
Looks like anyone can hit this endpoint and retrieve the whole user object based on any email?

Reply:
🤦‍♂️ thanks for the catch!